aicsimageio/tests/readers/test_glob_reader.py

```diff
@@ -68,6 +70,52 @@ def test_glob_reader_2d(tmp_path: Path) -> None:
     check_values(gr, reference)
 
 
+def test_index_alignment(tmp_path: Path) -> None:
+    # Testing case where user has passed in a list of files
+    # and a dataframe with non-continuous index
+
+    # use as_mm to have an easily available Indexer function
+    _ = make_fake_data_2d(tmp_path, as_mm=True)
+    filenames = np.array(list((tmp_path / "2d_images").glob("*.tif")))
+    # print(filenames)
+    indexer = pd.Series(filenames).apply(TiffGlobReader.MicroManagerIndexer)
+
+    # Keep only some of the Z
+    # more realistic case is eliminating everything after a given time point
+    # but this garuntees that our eliminated images will be embedded all through the
+    # order rather than just at the end
+    keep = indexer.Z < 5
+
+    indexer = indexer.loc[keep]
+
+    reader = TiffGlobReader(filenames[keep], indexer)
+
+    # check that there are no nans
+    # nans are a symptom of index misalignment
+    assert not reader._all_files.isnull().any().any()
+
+
+def test_glob_types(tmp_path: Path) -> None:
+    reference = make_fake_data_2d(tmp_path)
+    filenames = list((tmp_path / "2d_images").glob("*.tif"))
+
+    # list
+    gr = TiffGlobReader(list(filenames))
+    check_values(gr, reference)
+
+    # pd.Series
+    gr = TiffGlobReader(pd.Series(filenames))
+    check_values(gr, reference)
+
+    # np.array
+    gr = TiffGlobReader(np.array(filenames))
+    check_values(gr, reference)
+
+    with pytest.raises(TypeError):
+        # should throw a TypeError instead of an unboundlocal error
+        TiffGlobReader(2022)  # type: ignore
```

Comment:
Small nit, this could be a parametrized test! The parametrized version will give us errors for each failing type rather than stopping after the first failed check.
```suggestion
@pytest.mark.parametrize(
    "type_", [
        list,
        pd.Series,
        np.array, 
        # should throw a TypeError instead of an unboundlocal error
        pytest.param(bytes, marks=pytest.mark.raises(exception=TypeError)), 
    ]
)
def test_glob_types(type_, tmp_path: Path) -> None:
    reference = make_fake_data_2d(tmp_path)
    filenames = list((tmp_path / "2d_images").glob("*.tif"))

    gr = TiffGlobReader(type_(filenames))
    check_values(gr, reference)
```

Reply:
very slick :)